components/dropdown/DropdownMenu.vue

```diff
@@ -0,0 +1,101 @@
+<script setup lang="ts">
+import {
+  offset,
+  useFloating,
+  flip,
+  autoPlacement,
+  autoUpdate,
+  hide,
+} from "@floating-ui/vue";
+import type { Placement } from "@floating-ui/vue";
+import {
+  Menu,
+  MenuButton,
+  MenuItems,
+  TransitionRoot,
+  TransitionChild,
+} from "@headlessui/vue";
+
+const props = defineProps<{
+  placement?: Placement;
+}>();
+
+const buttonRef = ref<null | any>(null);
```

Comment:
Can we avoid using `any` here?

Reply:
yea I gave up trying to find the correct type 💀 but found it now